src/averageAges.js

```diff
@@ -36,8 +44,16 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
```

Comment:
You can simplify the filtering logic by combining it with the initial 'women' filtering. Use a single filter method with a condition that checks both if the person is female and if they have children.